test/e2e/framework/service_util.go

```diff
@@ -1522,3 +1664,55 @@ func CheckAffinity(jig *ServiceTestJig, execPod *v1.Pod, targetIP string, target
 	}
 	return true
 }
+
+// CreatePausePodDeployment creates a deployment for agnhost-pause pod running in different nodes
+func (j *ServiceTestJig) CreatePausePodDeployment(name, ns string, replica int32) *appsv1.Deployment {
+	immediate := int64(0)
+	pauseDeployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   name,
+			Labels: PauseDeploymentLabels,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replica,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: PauseDeploymentLabels,
+			},
+			Strategy: appsv1.DeploymentStrategy{
+				Type: appsv1.RollingUpdateDeploymentStrategyType,
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: PauseDeploymentLabels,
+				},
+				Spec: v1.PodSpec{
```

Comment:
Hmmmm. Was this here before? I don't recall it. Anti-affinity is not a guarantee, depends on scheduling availability. If we already are requiring two nodes, why not just use nodeSelector?

Reply:
>let's go with that - use anti-affinity but only 2 pods.

Done.